core-services/src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java

```diff
@@ -177,4 +178,44 @@ public void whenCreateTimelineForCompactedEventTypeThenException() throws Except
         timelineService.createTimeline("et1", "st1");
     }
 
+    @Test
+    public void whenRepartitioningEventTypeThenOk() {
+        final EventType eventType = buildDefaultEventType();
+
+        final Timeline t1 = Timeline.createTimeline(eventType.getName(), 1, null, "topic1", new Date());
+        t1.setSwitchedAt(new Date());
+        t1.setDeleted(true);
+
+        final Storage s1 = new Storage();
+        s1.setId("live");
+        s1.setType(Storage.Type.KAFKA);
+        final Timeline t2 = Timeline.createTimeline(eventType.getName(), 2, s1, "topic2", new Date());
+        t2.setSwitchedAt(new Date());
+        t2.setLatestPosition(new Timeline.KafkaStoragePosition(Lists.newArrayList(1l)));
+
+        final Storage s2 = new Storage();
+        s2.setId("backup");
+        s2.setType(Storage.Type.KAFKA);
+        final Timeline t3 = Timeline.createTimeline(eventType.getName(), 3, s2, "topic3", new Date());
+        t3.setSwitchedAt(new Date());
+        t3.setLatestPosition(new Timeline.KafkaStoragePosition(Lists.newArrayList(2l)));
+
+        final TopicRepository topicRepository1 = mock(TopicRepository.class);
+        final TopicRepository topicRepository2 = mock(TopicRepository.class);
+
+        when(eventTypeCache.getTimelinesOrdered(eventType.getName()))
+                .thenReturn(ImmutableList.of(t1, t2, t3));
+        when(topicRepositoryHolder.getTopicRepository(s1))
+                .thenReturn(topicRepository1);
+        when(topicRepositoryHolder.getTopicRepository(s2))
+                .thenReturn(topicRepository2);
+
+        timelineService.updateTimeLineForRepartition(eventType, 2);
+
+        Mockito.verify(topicRepository1, Mockito.times(1)).repartition("topic2", 2);
+        Mockito.verify(topicRepository2, Mockito.times(1)).repartition("topic3", 2);
+
+        Assert.assertEquals("[1, -1]", t2.getLatestPosition().toDebugString());
```

Comment:
it is not clear why you check latest position, if objection is to repartition. maybe check number of partitions before and after, all necessary calls were made. another option is to go with acceptance test with two storages, which is longer, but reliable.

Reply:
It is `TimelineService` responsibility to add the initial offset to each timeline for a newly added partition. Any active timeline that is not the latest one should hold the offsets of the prior Kafka topic.

eg:
 **before** repartition(topic with 1 partition), any active timeline should have the latest offset(s) from the time of timeline therefore `latestPosition=[1]`

**after** repartition(increase partition count by 1), all active timelines(beside the most recent one) should be patched with an initial offset eg `latestPosition=[1,-1]`

Not sure about the acceptance test, if you are referring to integration tests against an actual running system then we have to spawn 2 Kafka clusters, which I am not sure about it.

Reply:
the acceptance could you the same kafka, just create a separate storage, but lets proceed with what you have, the test covers the part was fixed 👍